api/src/main/java/org/openmrs/util/UpgradeUtil.java

```diff
@@ -62,68 +61,81 @@ public static Integer getConceptIdForUnits(String units) {
 				throw new APIException(e);
 			}
 		}
-		
+
 		throw new APIException("upgrade.settings.file.not.have.mapping", new Object[] { units });
 	}
-	
-	public static String getConceptUuid(Connection connection, int conceptId) throws SQLException {
 
-		try (PreparedStatement select = connection.prepareStatement("select uuid from concept where concept_id = ?")) {
+	public static String getConceptUuid(Connection connection, int conceptId) throws SQLException {
+		PreparedStatement select = connection.prepareStatement("select uuid from concept where concept_id = ?");
+		try {
 			select.setInt(1, conceptId);
-			
+
 			ResultSet resultSet = select.executeQuery();
 			if (resultSet.next()) {
 				return resultSet.getString(1);
 			} else {
 				throw new IllegalArgumentException("Concept not found " + conceptId);
 			}
 		}
+		finally {
```

Comment:
you need to discard your changes.

you can see the changes that introduced the try-with-resource at
https://github.com/openmrs/openmrs-core/commit/b99cdc85526c08e7f6da3f4aab5ae5666f184e63

from  https://issues.openmrs.org/browse/TRUNK-5031